src/NUnitFramework/framework/Internal/IndirectTestAssemblyAttribute.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+
+namespace NUnit.Framework.Internal
+{
+    /// <summary>
+    /// Causes the assembly to be skipped when autodiscovering tests.
+    /// Useful for tests-on-tests scenarios so that tools that run all tests in the solution do not execute the indirect tests directly.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Assembly)]
```

Comment:
If this is an attribute for tools to look for, why isn't it sufficient to just have the attribute? Why all the changes in the framework code itself?